lib/engine/game/g_18_india/step/corporate_sell_shares_company.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/corporate_sell_shares'
+require_relative '../../../step/share_buying'
+require_relative '../../../action/corporate_sell_shares'
+require_relative '../../../action/corporate_sell_company'
+
+module Engine
+  module Game
+    module G18India
+      module Step
+        class CorporateSellSharesCompany < Engine::Step::CorporateSellShares
+          include Engine::Step::ShareBuying
+
+          # update description
+          def description
+            'Corporate Sell Certificates to Market'
+          end
+
+          # Can sell any / all shares and companies, no restriction on purchases later
+          def actions(entity)
+            return [] unless entity == current_entity
+
+            actions = []
+            actions << 'corporate_sell_shares' if can_sell_any?(entity)
+            actions << 'corporate_sell_company' if can_sell_any_companies?(entity)
+            actions << 'pass' if actions.any?
+
+            actions
+          end
+
+          # ------ Sell Companies (new methods) ------
+
+          def can_sell_any_companies?(entity)
+            sellable_companies(entity).any?
+          end
+
+          def sellable_companies(entity)
+            return [] unless entity.corporation?
+
+            entity.companies.select { |c| c.type == :private || c.type == :bond }
+          end
+
+          def can_sell_company?(company)
+            return false unless company.company?
+            return false if company.owner != current_entity
+
+            company.type == :private || company.type == :bond
+          end
+
+          def process_corporate_sell_company(action)
+            company = action.company
+            corp = action.entity
+            raise GameError, "Cannot sell #{company.id}" unless can_sell_company?(company)
+
+            sell_company(corp, company, action.price, @game.bank)
+          end
+
+          def sell_price(company)
+            return 0 unless can_sell_company?(company)
+
+            company.value - @game.class::COMPANY_SALE_FEE
+          end
+
+          def sell_company(corp, company, price, bank)
+            company.owner = bank
+            bank.companies.push(company)
+            corp.companies.delete(company)
+            bank.spend(price, corp) if price.positive?
+            @log << "#{corp.name} sells #{company.name} to the Bank for #{@game.format_currency(price)}"
+          end
+
+          # ----- Sell Shares -----
+
+          # There are no restrictions on sales. Any amount can be sold to market
+          def can_sell?(_entity, bundle)
+            return unless bundle
+
+            true
+          end
```

Comment:
Is this override necessary?

I thought that `MARKET_SHARE_LIMIT` in the game class would allow for any amount to be sold to the market.
It looks like the only other functionality that's overridden from the base method is `bought?`, but since the sell step is before the bought step, won't that return `false` anyway?

Reply:
The main issue I needed to override in the base method was `entity != bundle.corporation`.  This would prevent the corporation from selling shares of itself.  The corp is allowed to buy it's share from the Market (but not IPO Rows).  This is the reason I refactored `corporation.ipo_owner` to be something other than `self`.